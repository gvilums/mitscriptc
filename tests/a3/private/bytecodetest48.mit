x = "a" + "b";
print(x);
print("should be 6 " + (3 * 2));
print("should be 3 " + (21 / 7));
print("should be true " + (5 > 3));
print("should be false " + (5 < 3));
print("should be true " + ("foo" == "foo"));