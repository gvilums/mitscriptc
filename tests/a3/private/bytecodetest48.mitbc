function
{
	functions =
	[
		function
		{
			functions = [],
			constants = [],
			parameter_count = 1,
			local_vars = [],
			local_ref_vars = [],
			free_vars = [],
			names = [],
			instructions = 
			[
			]
		},
		function
		{
			functions = [],
			constants = [],
			parameter_count = 0,
			local_vars = [],
			local_ref_vars = [],
			free_vars = [],
			names = [],
			instructions = 
			[
			]
		},
		function
		{
			functions = [],
			constants = [],
			parameter_count = 1,
			local_vars = [],
			local_ref_vars = [],
			free_vars = [],
			names = [],
			instructions = 
			[
			]
		}
	],
	constants = ["a", "b", "should be 6 ", 3, 2, "should be 3 ", 21, 7, "should be true ", 5, "should be false ", "foo"],
	parameter_count = 0,
	local_vars = [],
	local_ref_vars = [],
	free_vars = [],
	names = [x, print, input, intcast],
	instructions = 
	[
		load_func	0
		alloc_closure	0
		store_global	1
		load_func	1
		alloc_closure	0
		store_global	2
		load_func	2
		alloc_closure	0
		store_global	3
		load_const	0
		load_const	1
		add
		store_global	0
		load_global	1
		load_global	0
		call	1
		pop
		load_global	1
		load_const	2
		load_const	3
		load_const	4
		mul
		add
		call	1
		pop
		load_global	1
		load_const	5
		load_const	6
		load_const	7
		div
		add
		call	1
		pop
		load_global	1
		load_const	8
		load_const	9
		load_const	3
		gt
		add
		call	1
		pop
		load_global	1
		load_const	10
		load_const	9
		load_const	3
		swap
		gt
		add
		call	1
		pop
		load_global	1
		load_const	8
		load_const	11
		load_const	11
		eq
		add
		call	1
		pop
	]
}