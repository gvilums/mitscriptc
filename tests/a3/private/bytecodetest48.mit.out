ab
should be 6 6
should be 3 3
should be true true
should be false false
should be true true
